### ISSUE TITLE
Exit routes

### ISSUE THREAD

**Issue by aaronshaf** [thetalecrafter/middle-router#1]
Compare [page.exit](https://github.com/visionmedia/page.js#pageexitpath-callback-callback-)

**Comment by aaronshaf** [comment]
This is important for unsubscribe events.

**Comment by thetalecrafter** [comment]
Sounds awesome!

**Comment by aaronshaf** [comment]
I lightly suggest doing `await exit()` 

```javascript
async (ctx, next, exit) => {
  const unsubscribe = subscribe(render(mountNode)) // redux
  doAction()
  await next()
  doAnotherAction()
  await exit()
  unsubscribe()
}
```

**Comment by thetalecrafter** [comment]
implemented in 0.2.0
